js/map_utils.js

```diff
@@ -33,33 +33,40 @@ var stateStyle = {
   
 function addCentroids(map, countyCentroids) {
     
-  var geojson = topojson.feature(countyCentroids, countyCentroids.objects.foo);
-  
   scaleCircles
     .domain([
-              d3.min(geojson.features, function(d) { return d.properties[[activeCategory]]; }),
-              d3.max(geojson.features, function(d) { return d.properties[[activeCategory]]; })
+      d3.min(countyCentroids.features, function(d) { return d.properties[[activeCategory]]; }),
+      d3.max(countyCentroids.features, function(d) { return d.properties[[activeCategory]]; })
     ]);
   
   map.selectAll('county-point')
-    .data(geojson.features)
+    .data(countyCentroids.features)
     .enter()
     .append('circle')
     .classed('county-point', true)
-    ///////////////////
-    // Alaska COUNTYFIPS '016' coordinates currently in Canada
-    .filter(function(d) { return d.properties.COUNTYFIPS !== "016"; })
-    ///////////////////
     .sort(function(a,b) { 
       return d3.descending(a.properties[[activeCategory]], b.properties[[activeCategory]]);
     })
     .attr('fips', function(d) { return d.properties.GEOID; })
     .text(function(d) { return d.properties.GEOID; })
     .attr("cx", function(d) { 
-      return projection(d.geometry.coordinates)[0];
+      var proj = projection(d.geometry.coordinates);
+      if(!proj) {
+        console.log('bad projection:');
+        console.log(d);
+        console.log(projection(d.geometry.coordinates));
+        return 0;
+      } else {
+        return projection(d.geometry.coordinates)[0]; 
+      }
```

Comment:
this check is no longer technically necessary, but i bet it's not taking much processing time, and it could be useful if we go messing with the spatial data again (which i expect we will).

Reply:
So, `projection(d.geometry.coordinates)` was returning NULL, and `!proj` is TRUE for NULL? This isn't a critique of your implementation, I haven't used this syntax before so I wanted to understand it.

Reply:
Actual review: would it be more useful to print `d.geometry.coordinates` than just `d`?

Reply:
Yes, `!proj` is `true` for `null` - it's not the most obvious syntax, but I've seen it in several examples, so maybe it's standard, and I am not sure what other options exist.

i like printing d because then i can see both the coordinates and the properties, including state/county/fips, which helped me diagnose. i just expand the value in the console.